test/linter/test-descriptions.js

```diff
@@ -6,123 +6,108 @@ const { Logger } = require('./utils.js');
  */
 
 /**
- * @param {Identifier} apiData
- * @param {String} apiName
- * @param {Logger} logger
+ * Check for errors in the description of a specified statement's description and return whether there's an error and log as such
+ *
+ * @param {string} error_type The name of the error
+ * @param {string} name The name of the API method
+ * @param {Identifier} method The method's compat data
+ * @param {string} expected Expected description
+ * @param {Logger} logger The logger to output errors to
+ * @returns {void}
  */
-function hasValidConstrutorDescription(apiData, apiName, logger) {
-  const constructor = apiData[apiName];
-  if (
-    constructor &&
-    constructor.__compat.description !== `<code>${apiName}()</code> constructor`
-  ) {
-    logger.error(chalk`{red Incorrect constructor description for {bold ${apiName}()}}
-      {yellow Actual: {bold "${constructor.__compat.description || ''}"}}
-      {green Expected: {bold "<code>${apiName}()</code> constructor"}}`);
+function checkError(error_type, name, method, expected, logger) {
```

Comment:
1. Would suggest `check[Method/Feature]Description` instead of `checkError`, and renaming `error_type` to `rule_name` or similar.
2. In general, would it make sense to wrap all these functions in a `class` and access the logger with `this.logger`?
3. The function parameter casing is inconsistent in this PR, compare `processApiData(apiData, apiName, logger)` and `checkError(error_type, name, method, expected, logger)`.

Reply:
A class would be nice, however I'd like to hold off on doing any further refactoring until the next-gen linter, just to save extra effort in the meantime!

Reply:
Good point. 👍